src/deploy-cromwell-on-azure/KubernetesManager.cs

```diff
@@ -0,0 +1,261 @@
+        }
+
+        public async Task DeployHelmChartToCluster()
+        {
+           await ExecHelmProcess($"upgrade --install cromwellonazure ./scripts/helm --kubeconfig kubeconfig.txt --namespace {configuration.AksCoANamespace} --create-namespace");
+        }
+
+        public void UpdateHelmValues(string storageAccountName, string keyVaultUrl, string resourceGroupName, Dictionary<string, string> settings, IIdentity managedId)
+        {
+            var values = Yaml.LoadFromString<HelmValues>(Utility.GetFileContent("scripts", "helm", "values-template.yaml"));
+            values.Persistence["storageAccount"] = storageAccountName;
+            values.Persistence["keyVaultUrl"] = keyVaultUrl;
+            values.Persistence["keyVaultSecretName"] = Deployer.StorageAccountKeySecretName;
+            values.Config["resourceGroup"] = resourceGroupName;
+            values.Config["azureServicesAuthConnectionString"] = settings["AzureServicesAuthConnectionString"];
+            values.Config["applicationInsightsAccountName"] = settings["ApplicationInsightsAccountName"];
+            values.Config["cosmosDbAccountName"] = settings["CosmosDbAccountName"];
+            values.Config["batchAccountName"] = settings["BatchAccountName"];
+            values.Config["batchNodesSubnetId"] = settings["BatchNodesSubnetId"];
+            values.Config["coaNamespace"] = configuration.AksCoANamespace;
+            values.Config["disableBatchNodesPublicIpAddress"] = settings["DisableBatchNodesPublicIpAddress"];
+            values.Config["disableBatchScheduling"] = settings["DisableBatchScheduling"];
+            values.Config["usePreemptibleVmsOnly"] = settings["UsePreemptibleVmsOnly"];
+            values.Config["blobxferImageName"] = settings["BlobxferImageName"];
+            values.Config["dockerInDockerImageName"] = settings["DockerInDockerImageName"];
+            values.Config["batchImageOffer"] = settings["BatchImageOffer"];
+            values.Config["batchImagePublisher"] = settings["BatchImagePublisher"];
+            values.Config["batchImageSku"] = settings["BatchImageSku"];
+            values.Config["batchImageVersion"] = settings["BatchImageVersion"];
+            values.Config["batchNodeAgentSkuId"] = settings["BatchNodeAgentSkuId"];
+            values.Config["marthaUrl"] = settings["MarthaUrl"];
+            values.Config["marthaKeyVaultName"] = settings["MarthaKeyVaultName"];
+            values.Config["marthaSecretName"] = settings["MarthaSecretName"];
+            values.Identity["name"] = managedId.Name;
+            values.Identity["resourceId"] = managedId.Id;
+            values.Identity["clientId"] = managedId.ClientId;
+
+            if (!string.IsNullOrWhiteSpace(configuration.CustomTesImagePath))
+            {
+                values.Images["tes"] = configuration.CustomTesImagePath;
```

Comment:
CustomTesImagePath (and the other two) represents an export of locally built image (using _docker save_) to enable the scenario where developer build an image from source code locally and then uses it to deploy CoA, without having to upload the image to docker repository. Deployer copies the tar file to CoA host and then uses _docker load_ on the host. Is there a code that actually handles this scenario for AKS?

Is there code that replaces the image name in values-template.yaml with the current version?
For example, the value from the template, 
triggerservice: mcr.microsoft.com/cromwellonazure/triggerservice:2
needs to become something like:
triggerservice: mcr.microsoft.com/cromwellonazure/triggerservice:3.1.24567.2
This value is set by the build in file _env-02-internal-images.txt_ and can be overridden by the user by supplying --TriggerServiceImageName parameter. Same is true for Tes image. For Cromwell, user can set --CromwellVersion parameter.

Reply:
Thanks for clarifying this, I was getting CustomTesImagePath and TesImageName mixed up. I was using CustomTesImagePath for the scenario of passing Tes from a private ACR, fixed now to use TesImageName for this purpose.